File: ruby/squares/get_squares.js
var test = [1,4, 1, 16, 1, 10, 35, 22]
var result = get_square(test)

function get_square(array){
  var newArray = []
  for (var i = 0; i < array.length; i++) {
    var num = Math.sqrt(array[i])
    if (num % 1 === 0) {
      newArray.push(array[i])
    }
  }
  return uniq(newArray)
}

function uniq(a) {
    return a.sort().filter(function(item, pos, ary) {
        return !pos || item != ary[pos - 1];
    })
}

File: ruby/squares/get_squares.rb
require 'pry'


def get_square(array)
  newArray = []
  array.each do |num|
    var = Math.sqrt(num)

    if var % 1 == 0

      newArray << num
    end
  end

  return newArray.uniq.sort
end
test = [16 ,4, 1, 16, 1, 10, 35, 22]
result = get_square(test)
binding.pry
